addons/website_sale_wallet/security/ir.model.access.csv

```diff
@@ -0,0 +1,3 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_gift_card,access_gift_card,model_gift_card,base.group_user,1,1,1,1
+access_gift_card_apply_code,access_gift_card_apply_code,model_gift_card_apply_code,base.group_user,1,0,0,0
```

Comment:
don't forget to test all your flows with public user / portal user / demo user / admin

And always give less right than too much.
If needed use sudo when needed.